File: src/main/java/cz/solutions/cockroach/SalesReport.java
package cz.solutions.cockroach;

import lombok.Value;

import java.util.List;
import java.util.Map;

import static cz.solutions.cockroach.FormatingHelper.formatDouble;

@Value
public class SalesReport {
    List<PrintableSale> printableSalesList;
    double sellCroneValue;
    double sellDollarValue;
    double profitDolarValue;
    double recentProfitCroneValue;
    int totalAmount;
    double profitForTax;
    double sellCroneForTax;
    double buyCroneForTax;

    public Map<String,?> asMap(){
        return Map.of(
                "salesList", printableSalesList,
                "sellCroneValue", formatDouble(sellCroneValue),
                "sellDollarValue", formatDouble(sellDollarValue),
                "profitDolarValue", formatDouble(profitDolarValue),
                "profitRecentCroneValue", formatDouble(recentProfitCroneValue),
                "totalAmount", totalAmount,
                "profitForTax",profitForTax>0 ?  formatDouble(profitForTax):""
        );
    }
}

File: src/main/kotlin/cz/solutions/cockroach/SalesReport.kt
package cz.solutions.cockroach

data class SalesReport(
    val printableSalesList: List<PrintableSale>,
    val sellCroneValue: Double,
    val recentSellCroneValue: Double,
    val sellDollarValue: Double,
    val profitCroneValue: Double,
    val recentProfitCroneValue: Double,
    val totalAmount: Double,
    val buyDollarValue: Double,
    val profitForTax: Double,
    val buyCroneValue: Double,
    val recentBuyCroneValue: Double,
    val sellCroneForTax: Double,
    val buyCroneForTax: Double
)